searchkit/search.py

```diff
@@ -977,17 +978,27 @@ def _run_search(self, fd):
         self.stats.reset()
         sequence_results = SequenceSearchResults()
         search_ids = set([s.id for s in self.search_defs])  # noqa, pylint: disable=R1718
+        search_tags = set([s.tag for s in self.search_defs]) # noqa, pylint: disable=R1718
         offset = self.constraints_manager.apply_global(search_ids, fd)
-        log.debug("starting search of %s (offset=%s, pos=%s)", fd.name, offset,
-                  fd.tell())
+        log.debug("starting search of %s (offset=%s, pos=%s) for tags [%s]",
+                  fd.name, offset, fd.tell(), search_tags)
+
+        # Get the total file size (needed for calculating the search progress)
+        before_pos = fd.tell()
+        fd.seek(0, 2)
+        eof_offset = fd.tell()
+        fd.seek(before_pos)
+
         runnable = {s.id: _runnable
                     for s, _runnable in self.search_defs.items()}
         ln = 0
+        filename_short = Path(*Path(fd.name).parts[-2:])
         # NOTE: line numbers start at 1 hence offset + 1
         for ln, line in enumerate(fd, start=offset + 1):
             # This could be helpful to show progress for large files
             if ln % 100000 == 0:
-                log.debug("%s lines searched in %s", ln, fd.name)
+                log.debug("%s lines searched in %s (%g%%)",
```

Comment:
can we round the pcent to 3 dp e.g. {0:.3f}

Reply:
done:

```
2024-05-09 15:21:00,860 544680 DEBUG searchkit [-] 200000 lines searched in openstackrgxrh5t2/tmpry787c7r 55.455%)
2024-05-09 15:21:00,947 544681 DEBUG searchkit [-] 300000 lines searched in openstackrgxrh5t2/tmpcqx172jh 82.786%)
2024-05-09 15:21:00,954 544680 DEBUG searchkit [-] 300000 lines searched in openstackrgxrh5t2/tmpry787c7r 82.786%)
```